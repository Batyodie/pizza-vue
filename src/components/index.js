import Button from "./Button/Button.vue";
import Card from "./Card/Card.vue";
import DropDown from "./DropDown/DropDown.vue";
import Logo from "./Logo/Logo.vue";
import ShopCart from "./ShopCart/ShopCart.vue";

export { Button, Card, DropDown, Logo, ShopCart };
