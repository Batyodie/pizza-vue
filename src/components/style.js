import BtnTag from "@/components/Button/Button.css";
import Button from "@/components/Button/Button.css";
import DropDownStyle from "@/components/DropDown/DropDown.css";
import CardStyle from "@/components/Card/Card.css";
import LogoStyle from "@/components/Logo/Logo.css";
import ShopCartStyle from "@/components/ShopCart/ShopCart.css";
import BtnStyle from "@/components/Button/Button.css";

export {
  Button,
  BtnTag,
  DropDownStyle,
  CardStyle,
  LogoStyle,
  ShopCartStyle,
  BtnStyle
};
