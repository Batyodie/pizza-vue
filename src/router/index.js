import Vue from "vue";
import VueRouter from "vue-router";
import Home from "../views/Market";
import Basket from "../views/Basket/Basket";

Vue.use(VueRouter);

const routes = [
  {
    path: "/",
    name: "Home",
    component: Home,
  },
  {
    path: "/basket",
    name: "Basket",
    component: Basket,
  },
];

const router = new VueRouter({
  mode: "history",
  base: process.env.BASE_URL,
  hash: false,
  routes,
});

export default router;
