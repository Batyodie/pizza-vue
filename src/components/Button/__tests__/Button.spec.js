import { createLocalVue, mount } from "@vue/test-utils";
// dropdwons componens files
import { Button } from "@/components";

import VueRouter from "vue-router";
import router from "@/router";

const localVue = createLocalVue();
localVue.use(VueRouter);

describe("Unit tests for Button component", () => {
  // fake props
  const tagIndex = null;
  const icon = "/img/iconfinder.svg";
  const img = `<img src="${icon}" width="15" height="15" alt="icon basket" style="margin: 1px 0px 0px 13px;">`;
  const buttonText = "some any slots text";
  const wrapper = mount(Button, {
    localVue,
    router,
    propsData: {
      tagIndex
    },
    slots: {
      ButtonText: buttonText,
      Icon: img
    }
  });

  it("should initialize correctly Button component", () => {
    // correctly Vue instance
    expect(wrapper.exists()).toBeTruthy();
    // We expect the wrapper to be the Button component
    expect(wrapper.is(Button)).toBeTruthy();
    //  We expect the wrapper props to equal the fake props
    expect(wrapper.vm.tagIndex).toEqual(tagIndex);
    // add logo to data() component
  });

  it("should render correctly button component", () => {
    const btn = `<span class="Body">${buttonText}</span> ${img}`;
    // Expect wrapper html to have btn
    expect(wrapper.html()).toContain(btn);
    // Expect the wrapper to be the slots text render
    expect(wrapper.text(buttonText)).toBeTruthy();
    // Expect that render slot text in Body selector equal to buttonText
    expect(wrapper.find(".Body").text()).toEqual(buttonText);
    expect(wrapper.find("button > img").html()).toEqual(img);
  });

  it("Should match the snapshot Button component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("must return the event correctly", async () => {
    const btn = wrapper.find(".Btn");
    await btn.trigger("click");

    // expect checking that the event was triggered
    expect(wrapper.emitted("selectTag")).toBeTruthy();
    // expect checking that certain data was transmitted with the event
    expect(wrapper.emitted("selectTag")[0]).toEqual([tagIndex]);
  });
});
