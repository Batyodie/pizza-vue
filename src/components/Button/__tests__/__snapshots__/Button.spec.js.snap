// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for Button component Should match the snapshot Button component 1`] = `<button type="button" class="Btn"><span class="Body">some any slots text</span> <img src="/img/iconfinder.svg" width="15" height="15" alt="icon basket" style="margin: 1px 0px 0px 13px;"></button>`;
